weaviate/connect/authentication.py

```diff
@@ -94,14 +100,24 @@ def _get_session_user_pw(self, config: AuthClientPassword) -> OAuth2Session:
         return session
 
     def _get_session_client_credential(self, config: AuthClientCredentials) -> OAuth2Session:
+        scope: List[str] = self._default_scopes.copy()
+
+        # remove openid scopes from the scopes returned by weaviate (these are returned by default). These are not
+        # accepted by some providers for client credentials
+        if "openid" in scope:
+            scope.remove("openid")
+        if "email" in scope:
+            scope.remove("email")
```

Comment:
You could just filter out the unwanted scopes, that way the list is only traversed once:
```
scope = list(filter(lambda s: s != "openid" and s != "email", scope))
```

Reply:
done!